fvm/transactionInvoker.go

```diff
@@ -49,31 +49,42 @@ func (i TransactionInvoker) Process(
 
 	var modifiedSets programsCache.ModifiedSetsInvalidator
 	defer func() {
-		// based on the contract and frozen account updates we decide how to
-		// clean up the programs for failed transactions we also do the same as
-		// transaction without any deployed contracts
-		programs.Cleanup(modifiedSets)
-
 		if sth.NumNestedTransactions() > 1 {
-			err := sth.RestartNestedTransaction(nestedTxnId)
-			if err != nil {
+			if processErr == nil {
+				// This is a fvm internal programming error.  We forgot to
+				// call Commit somewhere in the control flow.  We should halt.
 				processErr = fmt.Errorf(
-					"cannot restart nested transaction: %w",
-					err,
-				)
+					"successfully executed transaction has unexpected " +
+						"nested transactions.")
```

Comment:
wonder if useful to log more context just for debugging purpose. For instance, `nextedTxnId`.

Reply:
the nestedTxnId is opaque (it holds a pointer) doesn't provide much debugging value.

Looks like we don't log the TxID in general.  I'll look into logging that consistent (in a separate PR)